lib/improver/tests/test_ensemble_copula_coupling_EnsembleCopulaCouplingUtilities.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for the
+`ensemble_copula_coupling.EnsemeblCopulaCouplingUtilities` class.
+"""
+import unittest
+
+from iris.coords import DimCoord
+from iris.cube import Cube
+from iris.tests import IrisTest
+import numpy as np
+
+from improver.ensemble_copula_coupling.ensemble_copula_coupling_utilities \
+    import create_percentiles, create_cube_with_percentiles
+from improver.tests.helper_functions_ensemble_calibration import (
+    set_up_temperature_cube, set_up_spot_temperature_cube,
+    _add_forecast_reference_time_and_forecast_period)
+
+
+class Test_create_cube_with_percentiles(IrisTest):
```

Comment:
Do we need a test for what happens if the data is the wrong shape and/or if the number of percentiles doesn't match the data.
You should maybe also be testing that the coordinates and metadata you want copied are copied.